src/server/json_family.cc

```diff
@@ -246,6 +246,44 @@ std::optional<JsonGetParams> ParseJsonGetParams(CmdArgParser* parser, Connection
   return parsed_args;
 }
 
+// This method makes a comparison of json considering their types
+// For example, 3 != 3.0 because json_type::int64_value != json_type::double_value
+bool JsonAreEquals(const JsonType& lhs, const JsonType& rhs) {
+  if (lhs.type() != rhs.type()) {
+    return false;
+  }
+  switch (lhs.type()) {
+    case json_type::array_value: {
+      if (lhs.size() != rhs.size()) {
+        return false;
+      }
+
+      auto r_it = rhs.array_range().begin();
+      return std::all_of(lhs.array_range().begin(), lhs.array_range().end(), [&](const auto& l_it) {
+        if (!JsonAreEquals(l_it, *r_it)) {
+          return false;
+        }
+        ++r_it;
+        return true;
+      });
```

Comment:
```suggestion
      for (auto r_it = ..., l_it = ...; r_it != end(); ++r_it, ++l_it) {
        if (!JsonAreEquals(l_it, *r_it)) {
          return false;
        }
      }
      return true;
```
Shorter and simpler, with no lambdas, I think it's more readable no?